packages/store/ts/config/v2/store.ts

```diff
@@ -56,12 +55,15 @@ export type resolveStore<store> = evaluate<{
   readonly codegen: "codegen" extends keyof store ? resolveCodegen<store["codegen"]> : resolveCodegen<{}>;
 }>;
 
-export function resolveStore<const store>(store: store): resolveStore<store> {
-  validateStore(store);
-
+export function resolveStore<const store extends StoreInput>(store: store): resolveStore<store> {
   return {
     tables: resolveTables(
-      mapObject(store.tables ?? {}, (table) => mergeIfUndefined(table, { namespace: store.namespace })),
+      Object.fromEntries(
+        Object.entries(store.tables ?? {}).map(([tableKey, table]) => {
+          const key = store.namespace ? `${store.namespace}__${tableKey}` : tableKey;
```

Comment:
not sure if we wanna lean on this here but we have `resourceToLabel` helper

Reply:
normally i would be down for that but in this particular case i feel like keeping the type and runtime implementation as similar as possible would be better